### ISSUE TITLE
0.3.11 release

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#2162]
ipfs version 0.3.11 has been tagged as a release canidate. This will be the final release before 0.4.0. Once 0.3.11 is shipped, 0.4.0 will merge into master. This means we need to make sure a few things happen:

- [x] 0.3.11 feedback
  - anything people want fixed before the 0.4.0 merge should be brought up now, not everything can be addressed, but major issues should receive some priority.
- [x] migrations testing
  - @chriscool is doing a great job setting up CI for the fs-repo-migrations, once we get that working I will feel more confident in the migrations efficacy.
- [ ] ipfs distributions page shipped
  - this one is being blocked by me currently, but before we drop 0.4.0, users need an easy way to get access to the new versions and the migrations binary (without requiring them to have go installed)
- [ ] discussion about migration path
  - the biggest issue here is figuring out what network `ipfs.io` resolves through. I don't want a lot of bikeshedding here, so i'm thinking that a good way forward will be to have `v04.ipfs.io` for now point to nodes on the dev0.4.0 network, and have both `ipfs.io` and `v03.ipfs.io` point to the current network. Once v0.4.0 is officially released, we will make `ipfs.io` point to the v04 network.

**Comment by Kubuxu** [comment]
It would be great to get #2116 in thus allowing to use sub-segements in dnslinks.

**Comment by diasdavid** [comment]
js-ipfs-api tests pass 🌟

```bash
» npm test

> ipfs-api@2.10.1 test /Users/david/code/protocol-labs/js-ipfs-api
> gulp test

[15:14:58] Using gulpfile ~/code/protocol-labs/js-ipfs-api/gulpfile.js
[15:14:58] Starting 'test'...
[15:14:58] Starting 'test:node'...
[15:14:58] Starting 'daemons:start'...
  ipfs init done - (bootstrap and mdns off) - b
  ipfs init done - (bootstrap and mdns off) - c
  ipfs init done - (bootstrap and mdns off) - a
[15:15:00] Finished 'daemons:start' after 1.91 s
[15:15:00] Starting 'mocha'...


  ipfsAPI constructor tests
    parameter permuations
      ✓ opts
      ✓ mutliaddr, opts
      ✓ host, port
      ✓ host, port, opts

  ipfsAPI request tests
    requestAPI
      ✓ excludes port from URL if config.port is falsy
      ✓ includes port in URL if config.port is truthy

  .add
    ✓ add file
    ✓ add buffer
    ✓ add BIG buffer (217ms)
    ✓ add path (48ms)
    ✓ add a nested dir
    ✓ add stream (39ms)
    ✓ add url (673ms)

  .block
    ✓ block.put
    ✓ block.get

  .cat
    ✓ cat
    ✓ cat BIG file (141ms)

  .commands
    ✓ lists commands

  .config
    ✓ .config.{set, get}
    ✓ .config.show
    ✓ .config.replace

  .dht
    ✓ returns an error when getting a non-existent key from the DHT
    ✓ puts and gets a key value pair in the DHT
    ✓ .dht.findprovs

  .diag
    ✓ .diag.net
    ✓ .diag.sys

  .id
    ✓ id

  API
    ✓ has the api object

  .log
    ✓ .log.tail (1171ms)

  .name
    ✓ .name.publish
    ✓ .name.resolve

  .object
    ✓ object.put
    ✓ object.get
    ✓ object.data
    ✓ object.stat
    ✓ object.links
    ✓ object.patch

  .pin
    ✓ .pin.add
    ✓ .pin.list
    ✓ .pin.remove

  .ping
    ✓ ping another peer (1008ms)

  .refs
    ✓ refs

  .send
    - used by every command

  .swarm
    ✓ .swarm.peers
    ✓ .swarm.connect

  .update (currently disabled, wait for IPFS 0.4.0)
    - .update.apply
    - .update.check
    - .update.log

  .version
      - running against version 0.3.11-dev
    ✓ checks the version


  45 passing (4s)
  4 pending

[15:15:04] Finished 'mocha' after 3.72 s
[15:15:04] Starting 'daemons:stop'...
[15:15:04] Finished 'daemons:stop' after 17 ms
[15:15:04] Finished 'test:node' after 5.65 s
[15:15:04] Starting 'test:browser'...
[15:15:04] Starting 'daemons:start'...
  ipfs init done - (bootstrap and mdns off) - a
  ipfs init done - (bootstrap and mdns off) - b
  ipfs init done - (bootstrap and mdns off) - c
[15:15:06] Finished 'daemons:start' after 1.71 s
[15:15:06] Starting 'karma'...

START:
12 01 2016 15:15:10.729:INFO [karma]: Karma v0.13.19 server started at http://localhost:9876/
12 01 2016 15:15:10.734:INFO [launcher]: Starting browser Chrome
12 01 2016 15:15:11.676:INFO [Chrome 47.0.2526 (Mac OS X 10.11.3)]: Connected on socket /#daiqOfYH1c4OPTlZAAAA with id 89449482
  .add
    ✔ add file
    ✔ add buffer
    ✔ add BIG buffer
    ✔ add path
    ✔ add a nested dir
    ✔ add stream
    ✔ add url
  .block
    ✔ block.put
    ✔ block.get
  .cat
    ✔ cat
    ✔ cat BIG file
  .commands
    ✔ lists commands
  .config
    ✔ .config.{set, get}
    ✔ .config.show
    ✔ .config.replace
  .dht
    ✔ returns an error when getting a non-existent key from the DHT
    ✔ puts and gets a key value pair in the DHT
    ✔ .dht.findprovs
  .diag
    ✔ .diag.net
    ✔ .diag.sys
  .id
    ✔ id
  API
    ✔ has the api object
  .log
    ✔ .log.tail
  .name
    ✔ .name.publish
    ✔ .name.resolve
  .object
    ✔ object.put
    ✔ object.get
    ✔ object.data
    ✔ object.stat
    ✔ object.links
    ✔ object.patch
  .pin
    ✔ .pin.add
    ✔ .pin.list
    ✔ .pin.remove
  .ping
    ✔ ping another peer
  .refs
    ✔ refs
  .send
    ✖ used by every command (skipped)
  .swarm
    ✔ .swarm.peers
    ✔ .swarm.connect
  .update (currently disabled, wait for IPFS 0.4.0)
    ✖ .update.apply (skipped)
    ✖ .update.check (skipped)
    ✖ .update.log (skipped)
LOG: '      - running against version', '0.3.11-dev'
  .version
    ✔ checks the version
  ipfsAPI constructor tests
    parameter permuations
      ✔ opts
      ✔ mutliaddr, opts
      ✔ host, port
      ✔ host, port, opts
  ipfsAPI request tests
    requestAPI
      ✔ excludes port from URL if config.port is falsy
      ✔ includes port in URL if config.port is truthy

Finished in 24.899 secs / 24.784 secs

SUMMARY:
✔ 45 tests completed
ℹ 4 tests skipped
[15:15:37] Finished 'karma' after 31 s
[15:15:37] Starting 'daemons:stop'...
[15:15:37] Finished 'daemons:stop' after 16 ms
[15:15:37] Finished 'test:browser' after 33 s
[15:15:37] Finished 'test' after 38 s
```

**Comment by Dignifiedquire** [comment]
New webui PR incoming soon

**Comment by Dignifiedquire** [comment]
@diasdavid is the header stuff for the api fixed now?

**Comment by diasdavid** [comment]
Which of the headers stuff? The commit that was merged after 0.3.10 released?

**Comment by Dignifiedquire** [comment]
@diasdavid yes

**Comment by diasdavid** [comment]
@Dignifiedquire It's there :) https://github.com/ipfs/go-ipfs/blob/master/commands/http/handler.go#L260-L263

**Comment by Dignifiedquire** [comment]
@diasdavid schweet

**Comment by whyrusleeping** [comment]
:ship: :shipit: :+1:
